### ISSUE TITLE
step.esa.int internal server error causes script to halt

### ISSUE THREAD

**Issue by scottstanie** [scottstanie/sentineleof#12]
Probably need to 
1. rate limit better for this server to be nicer
2. have a retry option for a 500
3. definitely not halt the rest when doing a mass download. just print a warning at the end for failed files

```bash
Traceback (most recent call last):
  File "/home/scott/miniconda3/envs/mapping/bin/eof", line 33, in <module>
    sys.exit(load_entry_point('sentineleof', 'console_scripts', 'eof')())
  File "/home/scott/miniconda3/envs/mapping/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/scott/miniconda3/envs/mapping/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/scott/miniconda3/envs/mapping/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/scott/miniconda3/envs/mapping/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/scott/repos/sentineleof/eof/cli.py", line 48, in cli
    download.main(
  File "/home/scott/repos/sentineleof/eof/download.py", line 329, in main
    return download_eofs(
  File "/home/scott/repos/sentineleof/eof/download.py", line 87, in download_eofs
    cur_filenames = result.get()
  File "/home/scott/miniconda3/envs/mapping/lib/python3.8/multiprocessing/pool.py", line 771, in get
    raise self._value
  File "/home/scott/miniconda3/envs/mapping/lib/python3.8/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/home/scott/repos/sentineleof/eof/download.py", line 222, in _download_and_write
    response.raise_for_status()
  File "/home/scott/miniconda3/envs/mapping/lib/python3.8/site-packages/requests/models.py", line 941, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: http://step.esa.int/auxdata/orbits/Sentinel-1/POEORB/S1B/2017/11/S1B_OPER_AUX_POEORB_OPOD_20171130T110835_V20171109T225942_20171111T005942.EOF.zip```

**Comment by scottstanie** [comment]
Hmm, well this actually turned into a `Connection Refused` problem... I appear to be blocked from the step.esa.int server from the workstation that I ran this from?

```python
In [2]: requests.get('http://step.esa.int/auxdata/orbits/Sentinel-1/POEORB/S1A/2017/02/')
---------------------------------------------------------------------------
ConnectionRefusedError                    Traceback (most recent call last)
~/miniconda3/envs/mapping/lib/python3.8/site-packages/urllib3/connection.py in _new_conn(self)
    158         try:
--> 159             conn = connection.create_connection(
    160                 (self._dns_host, self.port), self.timeout, **extra_kw

~/miniconda3/envs/mapping/lib/python3.8/site-packages/urllib3/util/connection.py in create_connection(address, timeout, source_address, socket_options)
     83     if err is not None:
---> 84         raise err
     85

~/miniconda3/envs/mapping/lib/python3.8/site-packages/urllib3/util/connection.py in create_connection(address, timeout, source_address, socket_options)
     73                 sock.bind(source_address)
---> 74             sock.connect(sa)
     75             return sock

ConnectionRefusedError: [Errno 111] Connection refused```

From my local computer, I can conect fine

```python
>>> requests.get('http://step.esa.int/auxdata/orbits/Sentinel-1/POEORB/S1A/2017/02/')
<Response [200]>
```

It's possible that it's just some networking issue that came up on my workstation.. but it seems like that IP was put on some list of IPs to refuse connections. 
If it's the case that the STEP team doesn't want this endpoint accessed and are blocking people, I'll definitely work this week to merge in one of the new API branches

**Comment by scottstanie** [comment]
ah, well here you go:
https://forum.step.esa.int/t/connection-refused-when-downloading-dem/29744/4

`The maximum number of allowed connections at a time from a singe IP address is 10.`

They are saying in the forums they STEP servers "are a last resort". 

@avalentino  I'll both limit the number of default connections (which is 20 right now, why I got banned in the first place), and merge my branch in while we finish up the `sentinelsat` version of it.

**Comment by scottstanie** [comment]
Fixed in #13
